### ISSUE TITLE
Expo 44 Beta Update Results in AppRegistry.registerComponent error

### ISSUE THREAD

**Issue by CodingItWrong** [expo/expo#15523]
### Summary

I followed the instructions in the [Expo 44 beta blog post](https://blog.expo.dev/expo-sdk-44-beta-is-now-available-75d9751b0a18) to update my app. It ran fine on web, but when I attempted to run it on iOS I get two errors, one of which is "Invariant Violation: "main" has not been registered."

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

iOS

### SDK Version (managed workflow only)

44.0.0-beta.1

### Environment

```
Expo CLI 4.13.0 environment info:
    System:
      OS: macOS 12.0.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 16.13.1 - ~/.nvm/versions/node/v16.13.1/bin/node
      Yarn: 1.22.17 - ~/.nvm/versions/node/v16.13.1/bin/yarn
      npm: 8.2.0 - ~/.nvm/versions/node/v16.13.1/bin/npm
      Watchman: 2021.11.15.00 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.11.2 - /Users/josh/.rbenv/shims/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.0.1, iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0
    IDEs:
      Xcode: 13.1/13A1030d - /usr/bin/xcodebuild
    npmPackages:
      expo: ^44.0.0-beta.1 => 44.0.0-beta.1
      react: 17.0.1 => 17.0.1
      react-dom: 17.0.1 => 17.0.1
      react-native: 0.64.3 => 0.64.3
      react-native-web: 0.17.1 => 0.17.1
    npmGlobalPackages:
      expo-cli: 4.13.0
    Expo Workflow: managed
```

### Reproducible demo

This issue is specifically about updating a project, so the reproduction seems like it needs to be a link to the project. Here is a link to a branch where you can see this occur: CodingItWrong/surely-expo#68

Steps to reproduce:

- yarn install
- yarn start
- run on iOS simulator

**Comment by CodingItWrong** [comment]
Full error output is:

```
Invariant Violation: "main" has not been registered. This can happen if:
* Metro (the local dev server) is run from the wrong folder. Check if Metro is running, stop it and restart it in the current project.
* A module failed to load due to an error and `AppRegistry.registerComponent` wasn't called.
at node_modules/react-native/Libraries/Core/ExceptionsManager.js:104:6 in reportException
at node_modules/react-native/Libraries/Core/ExceptionsManager.js:172:19 in handleException
at node_modules/react-native/Libraries/Core/setUpErrorHandling.js:24:6 in handleError
at node_modules/@react-native/polyfills/error-guard.js:49:36 in ErrorUtils.reportFatalError
```

**Comment by CodingItWrong** [comment]
The other error that occurs on app start is https://github.com/expo/expo/issues/15522

**Comment by brentvatne** [comment]
this could possibly be caused by the reanimated 2 error. let me know if this doesn't resolve, and use [this technique](https://github.com/expo/fyi/blob/main/manual-debugging.md#2-narrow-down-the-specific-changes-that-cause-your-error) to try to find the source then create a new issue.

**Comment by CodingItWrong** [comment]
This issue was in fact fixed by the fix to #15522, yes.
